server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/GroupRef.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.usergroups.ws;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class GroupRef {
+
+  private static final int NULL_ID = -1;
+
+  private final long id;
+  private final String organizationUuid;
+  private final String name;
+
+  private GroupRef(long id, @Nullable String organizationUuid, @Nullable String name) {
+    this.id = id;
+    this.organizationUuid = organizationUuid;
+    this.name = name;
+  }
+
+  /**
+   * @return {@code true} if id is defined and {@link #getId()} can be called. If {@code false}, then
+   *   the couple {organizationUuid, name} is defined and the methods {@link #getOrganizationUuid()}/{@link #getName()}
+   *   can be called.
+   */
+  public boolean hasId() {
+    return id != NULL_ID;
+  }
+
+  /**
+   * @return the group id
+   * @throws IllegalStateException if {@link #getId()} is {@code false}
+   */
+  public long getId() {
+    checkState(hasId(), "Id is not present. Please see hasId().");
+    return id;
+  }
+
+  /**
+   * @return the organization UUID. Always present, even if default organization.
+   * @throws IllegalStateException if {@link #getId()} is {@code true}
+   */
+  public String getOrganizationUuid() {
+    checkState(!hasId(), "Organization is not present. Please see hasId().");
+    return organizationUuid;
+  }
+
+  /**
+   * @return the non-null group name.
+   * @throws IllegalStateException if {@link #getId()} is {@code true}
+   */
+  public String getName() {
+    checkState(!hasId(), "Name is not present. Please see hasId().");
+    return name;
+  }
+
+  /**
+   * @return a new {@link GroupRef} referencing a group by its id
+   */
+  public static GroupRef fromId(long id) {
+    checkArgument(id > NULL_ID, "Group id must be positive: %s", id);
+    return new GroupRef(id, null, null);
+  }
+
+  /**
+   * @param organizationUuid non-null UUID of organization
+   * @param name non-null name
+   */
+  public static GroupRef fromName(String organizationUuid, String name) {
+    return new GroupRef(NULL_ID, requireNonNull(organizationUuid), requireNonNull(name));
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    GroupRef groupRef = (GroupRef) o;
+    if (id != groupRef.id) {
+      return false;
+    }
+    if (organizationUuid != null ? !organizationUuid.equals(groupRef.organizationUuid) : groupRef.organizationUuid != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS864)